statistics/handle/handle.go

```diff
@@ -251,28 +267,47 @@ func (h *Handle) updateStatsCache(newCache statsCache) {
 	h.statsCache.Lock()
 	oldCache := h.statsCache.Load().(statsCache)
 	if oldCache.version <= newCache.version {
+		h.statsCache.memTracker.Consume(newCache.memUsage - oldCache.memUsage)
 		h.statsCache.Store(newCache)
 	}
 	h.statsCache.Unlock()
 }
 
 func (sc statsCache) copy() statsCache {
-	newCache := statsCache{tables: make(map[int64]*statistics.Table, len(sc.tables)), version: sc.version}
+	newCache := statsCache{tables: make(map[int64]*statistics.Table, len(sc.tables)),
+		version:  sc.version,
+		memUsage: sc.memUsage}
 	for k, v := range sc.tables {
 		newCache.tables[k] = v
 	}
 	return newCache
 }
 
+func (sc statsCache) initMemoryUsage() {
```

Comment:
add a comment to note that it should be called after the tables and their stats are initilazed.